presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -448,6 +449,88 @@ public static Slice trim(@SqlType("varchar(x)") Slice slice)
         return SliceUtf8.trim(slice);
     }
 
+    @Description("remove the longest string containing only given characters from the beginning of a string")
+    @ScalarFunction("ltrim")
+    @LiteralParameters("x")
+    @SqlType("varchar(x)")
+    public static Slice leftTrim(@SqlType("varchar(x)") Slice slice, @SqlType(StandardTypes.VARCHAR) Slice charactersToTrim)
+    {
+        int[] codePointsToTrim = toCodePoints(charactersToTrim);
+        int start = leftTrimPosition(slice, codePointsToTrim);
+        return slice.slice(start, slice.length() - start);
+    }
+
+    private static int leftTrimPosition(@SqlType("varchar(x)") Slice slice, int[] codePointsToTrim)
+    {
+        int start = 0;
+        SLICE_LOOP:
+        while (start < slice.length()) {
+            int codePoint = getCodePointAt(slice, start);
+            for (int codePointToTrim : codePointsToTrim) {
+                if (codePoint == codePointToTrim) {
+                    start += lengthOfCodePoint(start);
+                    continue SLICE_LOOP;
```

Comment:
I don't generally have a problem with labels in Java. But I think in this particular case, I think it's better to write it with a boolean flag. Having an unconditional `break` at the end of a loop, and then skipping over it with a labeled `continue` seems convoluted.